src/deepsparse/benchmark_model/ort_engine.py

```diff
@@ -0,0 +1,304 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+from typing import Dict, Iterable, List, Optional, Tuple, Union
+
+import numpy
+
+from deepsparse.benchmark import BenchmarkResults
+from deepsparse.utils import (
+    get_input_names,
+    get_output_names,
+    model_to_path,
+    override_onnx_batch_size,
+    override_onnx_input_shapes,
+)
+
+
+try:
+    import onnxruntime
+
+    ort_import_error = None
+except Exception as ort_import_err:
+    onnxruntime = None
+    ort_import_error = ort_import_err
+
+try:
+    from sparsezoo import Zoo
+    from sparsezoo.objects import File, Model
+
+    sparsezoo_import_error = None
+except Exception as sparsezoo_err:
+    Zoo = None
+    Model = object
+    File = object
+    sparsezoo_import_error = sparsezoo_err
+
+try:
+    # flake8: noqa
+    from deepsparse.cpu import cpu_architecture
+except ImportError:
+    raise ImportError(
+        "Unable to import deepsparse python apis. "
+        "Please contact support@neuralmagic.com"
+    )
+
+__all__ = ["ORTEngine"]
+
+ARCH = cpu_architecture()
+NUM_CORES = ARCH.num_physical_cores
```

Comment:
should we be using available cores for this?  e.g. if this process has been numactl'd this will try to could use too many threads

Reply:
good call, we do the same in our engine. i assume numactl would deal with overriding everything but better safe than sorry